functional-tests/tests/test_anonymizer.py

```diff
@@ -1,92 +1,63 @@
-import json
-import os
-
 import pytest
+
+from common.assertions import equal_json_strings
 from common.methods import anonymize, anonymizers
 
 
 @pytest.mark.api
-def test_anonymize():
-    request_body = {
+def test_given_anonymize_called_with_valid_request_then_expected_valid_response_returned():
+    request_body = """
+    {
         "text": "hello world, my name is Jane Doe. My number is: 034453334",
-        "transformations": {"DEFAULT": {"type": "replace", "new_value": "ANONYMIZED"}},
+        "transformations": {
+            "DEFAULT": { "type": "replace", "new_value": "ANONYMIZED" },
+            "PHONE_NUMBER": { "type": "mask", "masking_char": "*", "chars_to_mask": 4, "from_end": true }
+        },
         "analyzer_results": [
-            {"start": 24, "end": 32, "score": 0.8, "entity_type": "NAME"}
-        ],
+            { "start": 24, "end": 32, "score": 0.8, "entity_type": "NAME" },
+            { "start": 24, "end": 28, "score": 0.8, "entity_type": "FIRST_NAME" },
+            { "start": 29, "end": 32, "score": 0.6, "entity_type": "LAST_NAME" },
+            { "start": 48, "end": 57,  "score": 0.95, "entity_type": "PHONE_NUMBER" }
+        ]
     }
+    """
 
     response_status, response_content = anonymize(request_body)
 
+    expected_response = "hello world, my name is ANONYMIZED. My number is: 03445****"
     assert response_status == 200
-    assert (
-        response_content
-        == "hello world, my name is ANONYMIZED. My number is: 034453334"
-    )
-
-
-@pytest.mark.api
-def test_anonymize_with_payload():
-    json_path = file_path("payload.json")
-    with open(json_path) as json_file:
-        content = json.load(json_file)
-
-    response_status, response_content = anonymize(content)
-
-    assert (
-        response_content
-        == "hello world, my name is ANONYMIZED. My number is: 03445****"
-    )
-    assert response_status == 200
-
-
-@pytest.mark.api
-def test_anonymize_with_payload_with_intersection_results():
-    json_path = file_path("intersection_payload.json")
-    with open(json_path) as json_file:
-        content = json.load(json_file)
-
-    response_status, response_content = anonymize(content)
-
-    assert (
-        response_content == "hello world, my name is <FULL_NAME><LAST_NAME>"
-        " My number is: <PHONE_NUMBER><SSN>"
-    )
-    assert response_status == 200
+    assert response_content == expected_response
 
 
 @pytest.mark.api
-def test_anonymize_api_fails_on_invalid_value_of_text():
-    json_path = file_path("payload.json")
-    with open(json_path) as json_file:
-        content = json.load(json_file)
-        content["text"] = ""
+def test_given_anonymize_called_with_empty_text_then_invalid_input_message_returned():
+    request_body = """
```

Comment:
Next test is with json files and this one is without I would keep it consistent. Take one valid json and explicitly do: json.text = ""
I think it will be more clear and use the same concept of json files.